mlrun/utils/helpers.py

```diff
@@ -1620,19 +1620,23 @@ def additional_filters_warning(additional_filters, class_name):
         )
 
 
-def merge_with_precedence(first_dict: dict, second_dict: dict) -> dict:
+def merge_with_precedence(
+    first_dict: dict, second_dict: dict, third_dict: dict = None
+) -> dict:
```

Comment:
```suggestion
def merge_with_precedence(
    first_dict: dict, *other_dicts tuple(dicts)
) -> dict:
```

merges N dictionaries..